tailwind.config.js

```diff
@@ -0,0 +1,169 @@
+module.exports = {
+  theme: {
+    backgroundOpacity: {
+      0: '0',
+      20: '0.2',
+      30: '0.3',
+      40: '0.4',
+      80: '0.8',
+      100: '1'
+    },
+    borderColor: theme => ({
+      1: theme("colors.border.1"),
+      2: theme("colors.border.2"),
+      3: theme("colors.border.3"),
+      4: theme("colors.border.4"),
+      5: theme("colors.border.5"),
+    }),
+    borderWidth: {
+      0: "0",
+      1: "1px",
+      3: "3px"
+    },
+    borderRadius: {
+      none: '0',
+      1: "var(--calcite-border-radius)",
+      2: "30px",
+      3: "50px",
+      half: '50%',
+      full: '100%',
+    },
+    boxShadow: {
+      1: {
+        default: "0 0 8px 0 rgba(0, 0, 0, 0.08)",
+        hover: "0 0 16px 0 rgba(0, 0, 0, 0.08)",
+        press: "0 0 8px 0 rgba(0, 0, 0, 0.16)"
+      },
+      2: {
+        default: "0 0 16px 0 rgba(0, 0, 0, 0.16)",
+        hover: "0 0 32px 0 rgba(0, 0, 0, 0.16)",
+        press: "0 0 16px 0 rgba(0, 0, 0, 0.32)",
```

Comment:
Probably a discussion for another time, but I find that I prefer a slight x/y offset on shadows. I am not entirely sure of my reason for this preference either :sweat_smile: . But, I think because its rare in nature to have a light source exactly perpendicular between you and an object (unless you are wearing a headlamp between your eyes). So, with an offset, the shadow seems natural at least to my eyes anyway.

If there's agreement, it might be cool to use just use the defaults (or some adaptation of it). Beyond the offset, they did some work to incorporate a "progressive negative spread to better convey an increase in elevation" (I found this [crash course](https://vimeo.com/441301137) on just what the heck that is to be insightful).

Reply:
/cc @bstifle

Reply:
I played around with this a bit. The following uses subset of the [defaults](https://tailwindcss.com/docs/box-shadow) (`default` - `lg`|`xl`). In keeping with what was already in place, I have attempted to preserve the two elevations. The pattern is a resting state `n`, hover state `n+1`, and pressed `n-1` (excluding level 1 - which is `n`, `n+1`, `n` respectively).

For example:

```css
.card {
  @apply shadow-md }
.card:hover {
  @apply shadow-lg }
.card:active {
  @apply shadow }
```
Or inline:
```html
<div class='h-24 w-24 shadow-md hover:shadow-lg active:shadow'></div>
```
Note, the `active:` variant would need to be added to the `variants` section e.g:
```js
  variants: {
    boxShadow: [ 'responsive', 'hover', 'focus', 'active' ]
  }
```

![Screenshot from 2020-08-31 13-31-51](https://user-images.githubusercontent.com/3051781/91748805-8ce62d80-eb8e-11ea-9848-48fb2b53070d.png)

Reply:
@jrowlingson - want to add this in?

Reply:
@macandcheese I've taken a stab at adding this in. I noticed that there is a case where the proposed shadows fell short. This is when there is a borderless, white, foreground over a white background. I added a new `shadow-borderless` series for this use case. 

Note, I've added a third elevation per the screenshot above and demoted tooltips into the new middle elevation. This can be reverted if desired.

Reply:
cc @bstifle could you check the above and see if adding another level works for our current elevation system with modal, tooltip, popover, dropdown, alert, notice, etc.

@jrowlingson - thanks for adding - would it be possible to use 4px increments for the box shadow values or is that just conforming to the scale for conforming sake (ie no values of 5px)?

Reply:
Good question, these shadow values are the tailwind defaults. I played around with snapping the offset + spread values to the existing spacing scale [here](https://codepen.io/jrowls/pen/QWNOLeY?editors=1100). The difference is minimal so it may make sense to just leave the defaults.

A shadow without a blur-radius is a simple rectangle with hard edges. I attempted to fit this rectangle to the scales. For example, `box-shadow: 0 12px 0px -4px gray` yield a rectangle below the target with a `0.5rem` height and `0.25rem` horizontal margin. Then the `blur-radius` applies a Gaussian blur centered at the edge of the rectangle. At `1px` the gradient is applied across `4px`, `2px` = `6px`, `3px` = `9px` and so on.

Since `box-shadow` does not affect spacing, and the math is a bit complex, it seems reasonable to leave it as is.

Reply:
Thanks for looking at that- it would definitely be nice to not just use Tailwind defaults if possible. But I'll let cc @bstifle decide that. I'm also unsure about the tooltip / popover level, seems a little arbitrary as far as what elements are level 2/3. Dropdown would probably be same as popover etc etc

Reply:
![Depth Light](https://user-images.githubusercontent.com/14083530/92148175-053c3100-edd1-11ea-80b2-12f08cb5c3ad.png)

@jrowlingson  We did a ton of work on depth a few months back. not sure if you are aware of this.

to adam's point, i dont think we need 3 shadow levels.

I like the idea of changing the press states to actually be physically pressed down. I also like the idea of moving the shadows down in the Y axis.

Reply:
while we are at it, we should probably just make the official corner radius 2px or 4px.

Reply:
^ Lets go to 4px for border radius value for things "above the plane"

Reply:
agreed

Reply:
@macandcheese @jrowlingson @paulcpederson @mitc7862 @asangma 

Ok yalls. Here's the new shadow proposal. I took jack's feedback and worked with it for quite a bit

I like the use of the double box shadows. gives that bottom edge a nice pump. More realistic and also... fåńčÿ!

These gotta work on #fff backgrounds, and the values jack gave would make the top edges blend in with white backgrounds. So I made sure each level has at lease a tiny bit of shadow on the top edge.

I really want to keep this down to 2 shadow levels, instead of 3 or 4.

Some of the box-shadow increments on here don't appear to be ultra logical, but after playing with this on multiple components, visually it just made more sense. Any feedback appreciated.

![alert notice dropdown](https://user-images.githubusercontent.com/14083530/93647789-e348ce00-f9ce-11ea-957b-6f3b8176611c.png)
![Depth Light-new](https://user-images.githubusercontent.com/14083530/93647790-e479fb00-f9ce-11ea-83ad-66979ea4d2ba.png)

Reply:
following up